### PR TITLE
feat: add QA skill with corrected CLI flag names and version baking

### PR DESCRIPTION
**Pull request by dwojtaszek**
## Release Notes

Adds a modular QA skill for functional testing of the Zipper CLI. The orchestrator (`.factory/skills/qa/`) analyzes git diffs and routes to the `qa-cli` sub-skill, which runs the binary with real arguments and verifies output files. Corrected flag names and build version baking discovered during the first QA run.

## Description

Initial QA skill setup with fixes applied from first test run:
- `--output` corrected to `--output-path` throughout all test flows
- `--chaos-percentage` corrected to `--chaos-mode` + `--chaos-amount`
- Absolute paths (`/tmp/...`) replaced with relative paths due to path traversal protection
- `--help` test removed (Zipper has no help flag)
- Build command now includes `-p:Version=qa-test` so `--version` output is meaningful
- Added 4 new Known Failure Modes documenting these discoveries

## Type of Change

- [x] New feature
- [ ] Bug fix
- [ ] Refactor
- [ ] Test coverage
- [ ] Documentation
- [ ] CI/Build
- [ ] Chore / Maintenance
- [ ] Dependency update

## Testing Done

- [x] Unit tests pass (`dotnet test src/Zipper.Tests/Zipper.Tests.csproj`)
- [x] E2E tests pass (`bash tests/run-tests.sh`)
- [x] Lint clean (`dotnet format --verify-no-changes src/`)
- [x] QA skill tested by running all 10 test flows (10/10 passed)

## Architecture

- [x] No deviation from the architecture diagrams

## Affected Requirements

None. This PR adds QA tooling only; no application behavior changes.

## Summary by CodeRabbit

* **New Features**
  * Added an end-to-end QA workflow for the Zipper CLI, including targeted change-based flow selection, quality gates, evidence capture, and failure-learning with suggested skill updates.

* **Documentation**
  * Added a detailed CLI QA test plan covering multiple test flows, artifact/format validations, and negative/boundary scenarios.
  * Added a QA report template with structured results and expandable evidence.

* **Chores**
  * Introduced QA skill configuration and completed installation progress tracking to standardize local execution and cleanup.

### CODE REVIEW

File: .factory/skills/qa/.install-progress.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  critical_flows:
+    answer: all_detected_flows_correct
+    flows:
+      - standard-mode
+      - loadfile-only
+      - loadfile-only-with-chaos
+      - production-set
+      - bates-numbering
+      - target-zip-size
+      - help-version
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply .factory/skills/qa/.install-progress.yaml]: @coderabbitai resolve

File: .factory/skills/qa/config.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+auth:
+  method: none
+  provider: none
+
+personas:
+  - name: default
+    description: 'CLI user running zipper with standard arguments'
+    credentials_source: none
+    test_focus: ['standard-mode', 'loadfile-only', 'production-set', 'target-zip-size', 'help-version']
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply .factory/skills/qa/config.yaml]: @coderabbitai resolve
